api/tests/integration/infrastructure/repositories/campaign-participation-repository_test.js

```diff
@@ -231,29 +232,38 @@ describe('Integration | Repository | Campaign Participation', () => {
     });
   });
 
-  describe('#findByUserId', () => {
+  describe('findLatestOngoingByUserId', () => {
 
     let userId;
 
-    beforeEach(async () => {
-      // given
-      userId = databaseBuilder.factory.buildUser({}).id;
-      _.times(2, () => {
-        databaseBuilder.factory.buildCampaignParticipation({ userId });
-      });
-      _.times(3, () => {
-        databaseBuilder.factory.buildCampaignParticipation({});
-      });
+    beforeEach(async() => {
+      userId = databaseBuilder.factory.buildUser().id;
       await databaseBuilder.commit();
     });
 
-    it('should return all the campaign-participation links to the given user', async () => {
-      // when
-      const foundCampaignParticipations = await campaignParticipationRepository.findByUserId(userId);
+    it('should retrieve the most recent campaign participations where the campaign is not archived', async () => {
+      const campaignId = databaseBuilder.factory.buildCampaign({ createdAt: new Date('2000-01-01T10:00:00Z'), archivedAt: null }).id;
+      const moreRecentCampaignId = databaseBuilder.factory.buildCampaign({ createdAt: new Date('2000-02-01T10:00:00Z'), archivedAt: null }).id;
+      const mostRecentButArchivedCampaignId = databaseBuilder.factory.buildCampaign({ createdAt: new Date('2001-03-01T10:00:00Z'), archivedAt: new Date('2000-09-01T10:00:00Z') }).id;
 
-      // then
-      expect(foundCampaignParticipations).to.have.a.lengthOf(2);
+      databaseBuilder.factory.buildCampaignParticipation({ userId, createdAt: new Date('2000-04-01T10:00:00Z'), campaignId: moreRecentCampaignId });
+      const expectedCampaignParticipationId = databaseBuilder.factory.buildCampaignParticipation({ userId, createdAt: new Date('2000-07-01T10:00:00Z'), campaignId }).id;
+      databaseBuilder.factory.buildCampaignParticipation({ userId, createdAt: new Date('2001-08-01T10:00:00Z'), campaignId: mostRecentButArchivedCampaignId });
+
+      databaseBuilder.factory.buildAssessment({ userId, campaignParticipationId: expectedCampaignParticipationId });
+      databaseBuilder.factory.buildAssessment({ userId, campaignParticipationId: expectedCampaignParticipationId });
+
+      await databaseBuilder.commit();
+
+      const [latestCampaignParticipation1, latestCampaignParticipation2, latestCampaignParticipation3] = await campaignParticipationRepository.findLatestOngoingByUserId(userId);
+
+      expect(latestCampaignParticipation1.createdAt).to.deep.equal(new Date('2000-07-01T10:00:00Z'));
+      expect(latestCampaignParticipation2.createdAt).to.deep.equal(new Date('2000-04-01T10:00:00Z'));
+      expect(latestCampaignParticipation1.assessments).to.be.instanceOf(Array);
+      expect(latestCampaignParticipation1.campaign).to.be.instanceOf(Campaign);
+      expect(latestCampaignParticipation3).to.be.undefined;
```

Comment:
est-ce que ça a vraiment de la valeur de tester ça ? Ce qu'on veut tester c'est plutôt que le tableau a une length de 2, non ?

Reply:
Alors la ligne 261 est essentielle, puisqu'on veut s'assurer qu'on remonte bien la bonne, la ligne 264 pour s'assurer que le tableau est de length 2, et les 2 autres pour s'assurer qu'on a bien les `withRequired` sinon le front casse, donc les 4 lignes me semblent importantes. 
C'est peut-être la valeur non conventionnelle de tester la length qui t'a perturbée ;)

Reply:
pardon j'ai pas précisé, je remettait juste en question la ligne 264 ^^

Reply:
Ce que je propose au lieu de `expect(latestCampaignParticipation3).to.be.undefined` c'est plutôt `expect(campaignParticipations).to.have.a.lengthOf(2)` :)

Reply:
done